cmd/osm-injector/osm-injector.go

```diff
@@ -0,0 +1,240 @@
+		log.Fatal().Msg("Error initializing generic event recorder")
+	}
+
+	// This ensures CLI parameters (and dependent values) are correct.
+	if err := validateCLIParams(); err != nil {
+		events.GenericEventRecorder().FatalEvent(err, events.InvalidCLIParameters, "Error validating CLI parameters")
+	}
+
+	stop := signals.RegisterExitHandlers()
+	_, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Start the default metrics store
+	metricsstore.DefaultMetricsStore.Start()
+
+	// Initialize Configurator to watch osm-config ConfigMap
+	cfg := configurator.NewConfigurator(kubeClient, stop, osmNamespace, osmConfigMapName)
+	configMap, err := cfg.GetConfigMap()
+	if err != nil {
+		log.Error().Err(err).Msgf("Error parsing ConfigMap %s", osmConfigMapName)
+	}
+	log.Info().Msgf("Initial ConfigMap %s: %s", osmConfigMapName, string(configMap))
+
+	// Initialize kubernetes.Controller to watch kubernetes resources
+	kubeController, err := k8s.NewKubernetesController(kubeClient, meshName, stop)
+	if err != nil {
+		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error creating Kubernetes Controller")
+	}
+
+	// Intitialize certificate manager/provider
+	certProviderConfig := providers.NewCertificateProviderConfig(kubeClient, kubeConfig, cfg, providers.Kind(certProviderKind), osmNamespace,
+		caBundleSecretName, tresorOptions, vaultOptions, certManagerOptions)
+
+	if caBundleSecretReadyChan, err := certProviderConfig.RegisterForCABundleSecretReadiness(); err != nil {
+		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error retrieving CA bundle secret")
+	} else {
+		// Wait for k8s CA bundle secret to be ready, required to initialize the certificate manager
+		<-caBundleSecretReadyChan
+	}
+
+	certManager, _, err := certProviderConfig.GetCertificateManager()
+	if err != nil {
+		events.GenericEventRecorder().FatalEvent(err, events.InvalidCertificateManager,
+			"Error initializing certificate manager of kind %s", certProviderKind)
+	}
+
+	// Initialize the sidecar injector webhook
+	if err := injector.NewMutatingWebhook(injectorConfig, kubeClient, certManager, kubeController, meshName, osmNamespace, webhookConfigName, stop, cfg); err != nil {
+		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error creating sidecar injector webhook")
+	}
+
+	/*
+	 * Initialize osm-injector's HTTP server
+	 */
+	httpServer := httpserver.NewHTTPServer(constants.OSMHTTPServerPort)
+	// Metrics
+	httpServer.AddHandler("/metrics", metricsstore.DefaultMetricsStore.Handler())
+	// Version
+	httpServer.AddHandler("/version", version.GetVersionHandler())
+	// Start HTTP server
+	err = httpServer.Start()
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Failed to start OSM metrics/probes HTTP server")
+	}
+
+	// Initialize the reconciler for the injector's MutatingWebhookConfiguration
+	if err := createControllerManagerForOSMResources(certManager); err != nil {
```

Comment:
It almost looks like (in another PR) we could rename this to something more specific, since it is for the MutatingWebhook resources and lives inside the injector.

Reply:
I agree, I copy pasted this from `osm-controller.go`, will rename.

Reply:
Fixed.